components/user-mgt/org.wso2.carbon.user.mgt/src/main/java/org/wso2/carbon/user/mgt/listeners/GroupManagementV2AuditLogger.java

```diff
@@ -0,0 +1,157 @@
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.user.mgt.listeners;
+
+import org.apache.commons.collections.CollectionUtils;
+import org.json.JSONArray;
+import org.json.JSONObject;
+import org.wso2.carbon.identity.central.log.mgt.utils.LoggerUtils;
+import org.wso2.carbon.identity.core.AbstractIdentityGroupOperationEventListener;
+import org.wso2.carbon.identity.core.util.IdentityCoreConstants;
+import org.wso2.carbon.user.core.UserStoreManager;
+import org.wso2.carbon.user.core.common.Claim;
+import org.wso2.carbon.user.core.common.Group;
+import org.wso2.carbon.utils.AuditLog;
+
+import java.util.List;
+
+import static org.wso2.carbon.identity.central.log.mgt.utils.LogConstants.UserManagement.ADDED_USERS_FIELD;
+import static org.wso2.carbon.identity.central.log.mgt.utils.LogConstants.UserManagement.ADD_GROUP_ACTION;
+import static org.wso2.carbon.identity.central.log.mgt.utils.LogConstants.UserManagement.DELETED_USERS_FIELD;
+import static org.wso2.carbon.identity.central.log.mgt.utils.LogConstants.UserManagement.DELETE_GROUP_ACTION;
+import static org.wso2.carbon.identity.central.log.mgt.utils.LogConstants.UserManagement.GET_GROUPS_OF_USERS_ACTION;
+import static org.wso2.carbon.identity.central.log.mgt.utils.LogConstants.UserManagement.GROUPS_FIELD;
+import static org.wso2.carbon.identity.central.log.mgt.utils.LogConstants.UserManagement.GROUP_NAME_FIELD;
+import static org.wso2.carbon.identity.central.log.mgt.utils.LogConstants.UserManagement.UPDATE_GROUP_NAME_ACTION;
+import static org.wso2.carbon.identity.central.log.mgt.utils.LogConstants.UserManagement.UPDATE_USERS_OF_GROUP_ACTION;
+import static org.wso2.carbon.identity.central.log.mgt.utils.LogConstants.UserManagement.USERS_FIELD;
+import static org.wso2.carbon.identity.central.log.mgt.utils.LoggerUtils.Target;
+import static org.wso2.carbon.identity.central.log.mgt.utils.LoggerUtils.isEnableV2AuditLogs;
+import static org.wso2.carbon.identity.central.log.mgt.utils.LoggerUtils.jsonObjectToMap;
+import static org.wso2.carbon.identity.central.log.mgt.utils.LoggerUtils.triggerAuditLogEvent;
+import static org.wso2.carbon.user.mgt.listeners.utils.ListenerUtils.getInitiatorId;
+
+/**
+ * This v2 audit logger logs the Group Management success activities.
+ */
+public class GroupManagementV2AuditLogger extends AbstractIdentityGroupOperationEventListener {
+
+    public boolean isEnable() {
+
+        if (super.isEnable()) {
+            return isEnableV2AuditLogs();
+        }
+        return false;
+    }
+
+    @Override
+    public boolean postAddGroup(String groupName, String groupId, List<String> userIds, List<Claim> claims,
+                                UserStoreManager userStoreManager) {
+
+        if (!isEnable()) {
```

Comment:
do we need this enable check?

Reply:
At this point we need to.
1. At kernel, we use **AbstractGroupOperationEventListener** which does not have the isEnabled check
2. **AbstractIdentityGroupOperationEventListener** which the logger extends from can not be used at kernel as it is from framework and hence introduce cyclic dependency.
3. This audit logger can be independently disabled. But audit loggers are enabled by default and we rely on the config to enable. We check if the AbstractIdentityGroupOperationEventListener is enabled and then proceed to check if **V2auditlogging** is enabled as well. We do this check at each method, to ensure unless this V2 is enabled, we should not log.